File: csharp/Testes/Teste4.cs
using System;
using System.Collections.Generic;

class Teste4
{
    public static void CalcularPercentualFaturamento()
    {
        var faturamentoPorEstado = new Dictionary<string, double>
        {
            { "SP", 67836.43 },
            { "RJ", 36678.66 },
            { "MG", 29229.88 },
            { "ES", 27165.48 },
            { "Outros", 19849.53 }
        };

        double totalFaturamento = 0;
        foreach (var valor in faturamentoPorEstado.Values)
        {
            totalFaturamento += valor;
        }

        Console.WriteLine("Percentual de representação por estado:");
        foreach (var estado in faturamentoPorEstado)
        {
            double percentual = (estado.Value / totalFaturamento) * 100;
            Console.WriteLine($"{estado.Key}: {percentual:F2}%");
        }
    }
}

File: javascript/teste 4.js
// 4) Dado o valor de faturamento mensal de uma distribuidora, detalhado por estado:
// • SP – R$67.836,43
// • RJ – R$36.678,66
// • MG – R$29.229,88
// • ES – R$27.165,48
// • Outros – R$19.849,53

// Escreva um programa na linguagem que desejar onde calcule o percentual de representação que cada estado teve dentro do valor total mensal da distribuidora.  

const faturamento = {
    "SP": 67836.43,
    "RJ": 36678.66,
    "MG": 29229.88,
    "ES": 27165.48,
    "Outros": 19849.53
};

function calculaPercentual(faturamento) {

    const total = Object.values(faturamento).reduce((acc, valor) => acc + valor, 0);

    const percentuais = Object.entries(faturamento).map(([estado, valor]) => {
        return { estado, percentual: ((valor / total) * 100).toFixed(2) + "%" };
    });

    console.log(`Faturameto total: R$${total}`);
    console.log("Percentual de representação de cada estado:");
    percentuais.forEach(({ estado, percentual }) => {
        console.log(`${estado}: ${percentual}`);
    });
}

calculaPercentual(faturamento);